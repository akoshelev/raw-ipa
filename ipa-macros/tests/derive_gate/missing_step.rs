mod common;

// this import is missing StepB which causes the error
use crate::common::{
    static_deserialize_state_map, static_reverse_state_map, static_state_map, Step, StepA,
    StepNarrow,
};
use ipa_macros::Gate;

#[derive(Gate)]
struct Compact(u16);

fn main() {
    let root = Compact(0);
    assert_eq!("root", root.as_ref());
}
