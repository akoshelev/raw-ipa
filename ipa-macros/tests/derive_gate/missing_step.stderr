error[E0412]: cannot find type `StepB` in this scope
  --> tests/derive_gate/missing_step.rs:10:10
   |
10 | #[derive(Gate)]
   |          ^^^^ not found in this scope
   |
   = note: this error originates in the derive macro `Gate` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this enum
   |
4  + use crate::common::StepB;
   |
